### PR TITLE
Update the page on known issues and limitations

### PR DESCRIPTION
**Pull request by Reshrahim**
Thank you for helping make the Radius documentation better!

**Please follow this checklist before submitting:**

- [ ] [Read the contribution guide](https://docs.radapp.dev/contributing/docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

### Summary
📝🔀🚧

This pull request enhances the `limitations.md` document by adding new information about resource naming and AWS compatibility issues with Radius Bicep, and by reorganizing the `rad CLI` limitations section.

> _Underscores of doom, non-idempotent rage_
> _Bicep your way through the AWS cage_
> _Limitations of the rad CLI, reorganized for clarity_
> _Read the document well, or face the error hell_

### Walkthrough
*  Document the limitation of using underscores in resource names and the workaround of using hyphens instead ([link](https://github.com/radius-project/docs/pull/801/files?diff=unified&w=0#diff-dbe4e906c77b4d35a11eca5955b5b5e4c672f17776180d05c3865a7f60a63a01L29-R65))
*  Reorganize the sections under `Containers` and `rad CLI` to improve readability and relevance ([link](https://github.com/radius-project/docs/pull/801/files?diff=unified&w=0#diff-dbe4e906c77b4d35a11eca5955b5b5e4c672f17776180d05c3865a7f60a63a01L57-R95))
*  Document the limitation of using non-idempotent AWS resources with Radius Bicep and the alternative of using idempotent resources or custom resources ([link](https://github.com/radius-project/docs/pull/801/files?diff=unified&w=0#diff-dbe4e906c77b4d35a11eca5955b5b5e4c672f17776180d05c3865a7f60a63a01L57-R95))

### CODE REVIEW

File: docs/content/reference/limitations.md [hunk 1]
```diff
@@ -26,13 +26,43 @@ This will be addressed further in a future release.
 
 A Radius environment allows you to specify Kubernetes as your compute platform, as well as specify the Kubernetes namespace in which Kubernetes objects are deployed. Additionally, you can override the namespace for a specific application using the [kubernetesNamespace extension.]({{< ref "application-schema#kubernetesNamespace" >}}). Currently, changing the namespace of an environment or application requires the application to be deleted and redeployed. If you need to change the namespace of an application, you can do so by deleting the application and/or environment and redeploying it with the new namespace.
 
+### Cannot use underscores in resource names
```
Comment (AaronCrawfis) [inline docs/content/reference/limitations.md]:
```suggestion
### Resource names cannot contain underscores (_)
```

File: docs/content/reference/limitations.md [hunk 2]
```diff
@@ -26,13 +26,43 @@ This will be addressed further in a future release.
...
+
+Using an underscore in a resource name will result in an error.
+
+As a workaround do not use underscores in resource names. Additional validation will be added in a future release to help warn against improperly formatted resource names.
+
+See [app name constraints]({{< ref "resource-schema.md#common-values" >}}) for more information.
+
+## rad CLI
+
+### Application and resource names are lower-cased after deployment
```
Comment (AaronCrawfis) [inline docs/content/reference/limitations.md:39]: Do we have a bug tracking this? 

cc/ @rynowak

**Review by AaronCrawfis — APPROVED** [review]
Overall looks good. Couple small things

File: docs/content/reference/limitations.md [hunk 3]
```diff
@@ -54,34 +84,12 @@ var stgSuffixes = az.environment().suffixes.storage
 
 This will be addressed in a future release when we change how the environmentId is passed into the file.
 
-## Containers
+### Radius Bicep AWS limitations
 
-## rad CLI
+Some of the [supported AWS resource types](/resource-schema/aws) are 'non-idempotent', this means that this resource type is assigned a primary identifier at deployment time and is currently not supported by Radius Bicep.
```
Comment (Reshrahim) [inline docs/content/reference/limitations.md]:
```suggestion
Some of the [AWS resource types](/resource-schema/aws) are 'non-idempotent', this means that this resource type is assigned a primary identifier at deployment time and is currently not supported by Radius Bicep.
```

File: docs/content/reference/limitations.md (hunk 2 above)
Comment (Reshrahim) [reply docs/content/reference/limitations.md:39]: couldn't find anything on the radius repo

Comment (AaronCrawfis) [reply docs/content/reference/limitations.md:39]: Could we open one up to track? This seems like a bug we should fix, either allowing multi-cased names or throwing an error if upper-cased letters are used.

**Comment by Reshrahim** [comment]
@AaronCrawfis - Should we highlight the Terraform Radius provider support as a known limitation?

**Comment by AaronCrawfis** [comment]
> @AaronCrawfis - Should we highlight the Terraform Radius provider support as a known limitation?

Nope. New features we haven't built yet aren't limitations. Things we've built that don't work the way we want them to are more of how we describe limitations.

**Review by AaronCrawfis — APPROVED** [review]

**Pull request merged**